Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -2558,6 +2574,83 @@ private final class TokenStreamCreator: SyntaxVisitor {
     // always parse the former as "4 +- 5".
     return true
   }
+
+  /// Appends the given node to the token stream without applying any formatting or printing tokens.
+  ///
+  /// - Parameter node: A node that is ignored by the formatter.
+  private func appendFormatterIgnored(node: Syntax) {
+    // The first line of text in the `verbatim` token is printed with correct indentation, based on
+    // the previous tokens. The leading trivia of the first token needs to be excluded from the
+    // `verbatim` token in order for the first token to be printed with correct indentation. All
+    // following lines in the ignored node are printed as-is with no changes to indentation.
+    var nodeText = node.description
+    if let firstToken = node.firstToken {
+      extractLeadingTrivia(firstToken)
+      let leadingTriviaText = firstToken.leadingTrivia.reduce(into: "") { $1.write(to: &$0) }
+      nodeText = String(nodeText.dropFirst(leadingTriviaText.count))
+    }
+
+    // The leading trivia of the next token, after the ignored node, may contain content that
+    // belongs with the ignored node. The trivia extraction that is performed for `lastToken` later
+    // excludes that content so it needs to be extracted and added to the token stream here.
+    if let next = node.lastToken?.nextToken, let trivia = next.leadingTrivia.first {
+      switch trivia {
+      case .lineComment, .blockComment:
+        trivia.write(to: &nodeText)
+        break
+      default:
+        // All other kinds of trivia are inserted into the token stream by `extractLeadingTrivia`
+        // when the relevant token is visited.
+        break
+      }
+    }
+
+    appendToken(.verbatim(Verbatim(text: nodeText, indentingBehavior: .firstLine)))
+  }
+
+  /// Returns whether the given trivia includes a directive to ignore formatting for the next node.
+  ///
+  /// - Parameter trivia: Leading trivia for a node that the formatter supports ignoring.
+  private func isFormatterIgnorePresent(inTrivia trivia: Trivia) -> Bool {
+     for piece in trivia {
+       switch piece {
+       case .lineComment(let text), .blockComment(let text),
+         .docLineComment(let text), .docBlockComment(let text):
+         if text.contains("swift-format-ignore") { return true }
```

Comment:
Can we be more precise here, perhaps with a prefix check (I believe we'll have to strip off the comment leading characters, though)? That way we allow this:

```swift
// swift-format-ignore
```

but not this:

```swift
// Do disable the formatter, write "// swift-format-ignore".
```

Also, let's remove the support for `docLineComment` and `docBlockComment`; we don't want people to put directives in their documentation. If they want to disable formatting on a documented decl, we should require this:

```swift
// swift-format-ignore
/// Blah blah blah
func foo() {
  poorlyFormattedCode()
}
```

Reply:
I've dropped doc comments and updated to specifically check for exactly "swift-format-ignore" (ignoring whitespace).